### DISCUSSION TITLE
[Feature Request] Support nested menu items in Dropdown component

### DISCUSSION THREAD

**Discussion by iGmainC** [heroui-inc/heroui#1464] [Feature requests]
### Is your feature request related to a problem? Please describe.

Currently, the Dropdown Menu component in NextUI 2.0 only supports a single level of menu items. This limits the flexibility of the component and makes it difficult to create complex menu structures. I would like to request support for nested menu items in the Dropdown component.

### Describe the solution you'd like

I would like to see the Dropdown component support nested menu items, allowing for the creation of multi-level menu structures. This would involve adding support for submenus within the Dropdown component, along with the ability to toggle the display of submenus on hover or click.

### Describe alternatives you've considered

One alternative I have considered is using a different component library that already supports nested menu items. However, I would prefer to continue using NextUI for its other features and the overall consistency it provides to my application's UI.

### Screenshots or Videos

[shadcn/ui](https://ui.shadcn.com/docs/components/dropdown-menu)

![image](https://github.com/nextui-org/nextui/assets/48195631/2e79185b-715b-48c2-8063-7effb5d13548)

[ant design](https://ant.design/components/dropdown)

![image](https://github.com/nextui-org/nextui/assets/48195631/65f3f067-cfb7-42c7-a07b-aea822d9308c)

------------------

![image](https://github.com/nextui-org/nextui/assets/48195631/d30d5291-feef-4c4c-b548-05f9ba9697b3)

https://primereact.org/tieredmenu/#basic

---------------------

![image](https://github.com/nextui-org/nextui/assets/48195631/33ff0222-0731-4235-b6eb-b3fb1feef545)

https://react.semantic-ui.com/collections/menu/

---------------

![image](https://github.com/nextui-org/nextui/assets/48195631/e77bf488-5e1a-4281-b324-7194175b8ea5)

https://rsuitejs.com/components/dropdown/

**Comment by ForrestDevs** [comment]
also looking for a feature as such

**Comment by artemavrin** [comment]
yes, we need sub menus in dropdown too

**Comment by hrithiqball** [comment]
needed this so so much. i thought it would support by default

**Reply by ersanyakit** [reply]
yes...

**Comment by sagarj0** [comment]
yes indeed it is needed, i thought by nesting another Select component would work, it didn't.

**Comment by dev-idblfs** [comment]
this is what we need from next UI,

**Comment by MohamedDanis** [comment]
please add this nested dropdown

**Comment by Sheikh-Srijon** [comment]
i have 30 dropdown menu items to choose from. Currently it takes the whole page, is there a way i can seelct a subset like 8 menu items to be visible? User can scroll vertically for the rest

**Comment by acsoto** [comment]
Do you know of any other component libraries that support this?

**Reply by iGmainC** [reply]
![image](https://github.com/nextui-org/nextui/assets/48195631/d30d5291-feef-4c4c-b548-05f9ba9697b3)

https://primereact.org/tieredmenu/#basic

---------------------

![image](https://github.com/nextui-org/nextui/assets/48195631/33ff0222-0731-4235-b6eb-b3fb1feef545)

https://react.semantic-ui.com/collections/menu/

---------------

![image](https://github.com/nextui-org/nextui/assets/48195631/e77bf488-5e1a-4281-b324-7194175b8ea5)

https://rsuitejs.com/components/dropdown/

**Reply by acsoto** [reply]
Thank you!

**Comment by pmaik** [comment]
Please add this feature.

**Comment by vmorales-2920** [comment]
Please add this feature, is really needed

**Comment by chihabhajji** [comment]
+1

**Comment by ariffirdausmk** [comment]
hope you can add this soon

**Comment by a-chacon** [comment]
+1

**Comment by JamesClow** [comment]
We removed Hero UI from our stack because it doesn't support this feature.

**Comment by aeokay** [comment]
I have an idea, you can nest popover with this code`<Popover placement="bottom-end">
            <PopoverTrigger>
                <Button variant="light" size="sm" isIconOnly>
                    <Ellipsis size={20} />
                </Button>
            </PopoverTrigger>
            <PopoverContent>
               ....
                    <Popover placement="left-end">
                        <PopoverTrigger>
                            <Button>Nest button</Button>
                        </PopoverTrigger>
                        <PopoverContent>
                            <div className='flex flex-col gap-y-2'>
                                content nest
                            </div>
                        </PopoverContent>
                    </Popover>
                </div>
            </PopoverContent>
        </Popover>`

**Reply by sapkra** [reply]
I just formatted it a bit and removed the second closing div.
```tsx
const MyNestedPopover = (
  <Popover placement="bottom-end">
    <PopoverTrigger>
      <Button variant="light" size="sm" isIconOnly>
        <Ellipsis size={20} />
      </Button>
    </PopoverTrigger>

    <PopoverContent>
      ....
      <Popover placement="left-end">
        <PopoverTrigger>
          <Button>Nest button</Button>
        </PopoverTrigger>
        <PopoverContent>
          <div className='flex flex-col gap-y-2'>content nest</div>
        </PopoverContent>
      </Popover>
      ....
    </PopoverContent>
  </Popover>
);
```

**Reply by sapkra** [reply]
I actually did the same with two nested Dropdowns but the second one is controlled by a hover state that is set by the parent DropdownItem. The DropdownTrigger of the nested dropdown is just a div with the label in it.

**Reply by sapkra** [reply]
It's broken in newer versions now.

**Comment by aliamiri1380** [comment]
I've provided such functionality inside this repo:
https://github.com/aliamiri1380/fast-heroui


https://github.com/user-attachments/assets/78e666e0-817e-4689-9738-a4a2bef47288

full code:

```js
import  Dropdown  from  './parts/hero_dropdown'

<Dropdown subPlacement={'right'} items={[
		{ 'type': 'item', 'content': 'New file', 'icon': 'file-plus' },
		{
			'type': 'sub', 'content': 'Copy link', 'icon': 'link', 'items': [
				{ 'type': 'item', 'content': 'Copy full link', 'icon': 'link-plus' },
				{ 'type': 'item', 'content': 'Copy short link', 'icon': 'link-plus', 'iconProps': { 'size': 'md' } },
				{
					'type': 'sub', 'content': 'Copy colorful link', 'icon': 'link-plus', 'items': [
						{ 'type': 'item', 'content': 'red', 'props': { 'color': 'danger' } },
						{ 'type': 'item', 'content': 'green', 'props': { 'color': 'success' } },
						{ 'type': 'item', 'content': 'blue', 'props': { 'color': 'primary' } },
					]
				},
			]
		},
		{ 'type': 'item', 'content': 'Edit file', 'icon': 'pencil' },
		{ 'type': 'item', 'content': 'Delete file', 'icon': 'trash', 'props': { 'color': 'danger' } },
	]}>
		<Button>Open Menu</Button>
	</Dropdown>
```

see the readme for more details

**Comment by raulcarlomagno** [comment]
+1

**Comment by hasan6034** [comment]
please bring this feature

**Comment by drosestudios** [comment]
Is there any plans to make this a reality?
